observability/WFLY-20296-add-support-for-more-opentelemetry-config-options.adoc

```diff
@@ -0,0 +1,140 @@
+---
+categories:
+- observability
+stability-level: community
+# Specify the Feature Development tracker issue for the feature.
+# This must be an issue tracked in https://github.com/orgs/wildfly/projects/7/views/1.
+# To create a Feature Development tracker issue, go to  https://github.com/wildfly/wildfly-proposals/issues/new/choose
+# and select 'Feature Development'
+issue: https://github.com/wildfly/wildfly-proposals/issues/683
+feature-team:
+ developer: jasondlee
+ sme:
+  -
+ outside-perspective:
+  -
+promotes:
+promoted-by:
+---
+= WFLY-20296 - Add support for more OpenTelemetry configuration options
+:author:            Jason Lee
+:email:             jasondlee@redhat.com
+:toc:               left
+:icons:             font
+:idprefix:
+:idseparator:       -
+
+== Overview
+
+While every aspect of OpenTelemetry is technically configurable via MP Telemetry and MP Config, there have been several requests for more direct support for the various options in the management model. The RFE will identify and implement the options most likely to be of general interest to WildFly users.
+
+=== User Stories
+
+- As an application developer, I would like greater control in how OpenTelemetry is configured in the server without having to add an external library (e.g., MicroProfile Config) to my deployment(s).
+
+== Issue Metadata
+
+- https://issues.redhat.com/browse/WFLY-20296[]
+
+=== Related Issues
+
+- https://issues.redhat.com/browse/WFLY-18290[]
+- https://issues.redhat.com/browse/WFLY-19919[]
+- https://issues.redhat.com/browse/JBEAP-27199[]
+
+=== Affected Projects or Components
+
+This change will affect only the OpenTelemetry subsystem
+
+=== Other Interested Projects
+
+=== Relevant Installation Types
+
+* Traditional standalone server (unzipped or provisioned by Galleon)
+* Managed domain
+* OpenShift Source-to-Image (S2I)
+* Bootable jar
+
+== Requirements
+
+* Additional configuration options should include:
+** General
+*** TLS server certificate (`otel.exporter.otlp.certificate`)
+*** TLS client certificate (`otel.exporter.otlp.client.certificate`)
```

Comment:
Please discuss with the security team how this kind of thing should be represented; i.e. if we should provide deeper integration with existing Elytron services.

Reply:
What type of data would attributes for these take?

In general we are trying to centralise configuration related to SSLContexts into the Elytron subsystem so users and administrators don't need to learn how to confiure TLS related resources independently on each separate subsystem,

Reply:
The value passed to the underlying library is the full path to the certificates (and keys) used by the OpenTelemetry Collector to configure TLS on its end. I guess it's possible that we switch the perspective here: rather than pointing WF to what the Collector is using to configure itself, we point the Collector to what WF is using to configure itself?

Reply:
Yes. It's a process that runs in another JVM (or even data center). In my head, the plan is currently to point the OpenTelemetry subsystem at the certificate and key files known to and use by the OpenTelemetry Collector. What I mean by the inversion is that rather than point from inside the app servers config (and FS area) to files owned/managed by the Collector, that could instead point the Collector to cert/key files owned and managed by WildFly. Perhaps I was being too cute. :P I was just thinking "out loud" about how a WildFly SSL context might used to configure an external service. Ultimately, it's going to come down to each system sharing one or more files, regardless of who "owns" those files.